iron-request.html

```diff
@@ -186,18 +183,16 @@
         return null;
       }
 
-      xhr.addEventListener('readystatechange', function () {
-        if (xhr.readyState === 4 && !this.aborted) {
-          this._updateStatus();
-
-          if (!this.succeeded) {
-            this.rejectCompletes(new Error('The request failed with status code: ' + this.xhr.status));
-            return;
-          }
+      xhr.addEventListener('loadend', function() {
```

Comment:
It's an event that's spec'd to fire _after_ `error`, `timeout`, etc. This is super useful, and moving to it from `readystatechange` to detect the end of the request fixes some other issues too.

I wrapped up this change in an independent PR, so I'll wait to land this timeout PR until after the loadend PR lands: https://github.com/PolymerElements/iron-ajax/pull/126